Lib/test/_test_multiprocessing.py

```diff
@@ -3824,6 +3824,44 @@ def test_shared_memory_SharedMemoryServer_ignores_sigint(self):
 
         smm.shutdown()
 
+    def test_shared_memory_SharedMemoryManager_cleanup_if_parent_dies(self):
+        # If a process that spawned a SharedMemoryManager was terminated, its
+        # manager process should notice it, unlink the resources the manager
+        # delivered, and shut down.
+        cmd = '''if 1:
+            import sys, time
+
+            from multiprocessing.managers import SharedMemoryManager
+
+
+            smm = SharedMemoryManager()
+            smm.start()
+            sl = smm.ShareableList([1, 2, 3])
+
+            sys.stdout.write(f'{sl.shm.name}\\n')
+            sys.stdout.flush()
+            time.sleep(100)
+        '''
+
+        p = subprocess.Popen([sys.executable, '-E', '-c', cmd],
+                             stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        shm_name = p.stdout.readline().decode().strip()
+        p.terminate()
+        start_time = time.monotonic()
+        deadline = start_time + 60
```

Comment:
10 seconds sounds enough, no?

Reply:
In case of slow CI workers one can sometimes see weird things on rare occasions.

99.9% of the time it will probably take less than 10ms (although I have not checked), this limit is just a timeout, so I think it's fine to keep a large value to make it extremely unlikely to get false positive failures on CI servers.

Reply:
There are seconds here not milliseconds.

Reply:
But, no strong feelings either.